config/webpack/loaders/devSiteCodeblock.js

```diff
@@ -0,0 +1,37 @@
+const path = require('path');
+const findSource = require('../loaderUtils/findSource');
+
+/**
+ * Load the contents of the file into the codeblock
+ * This loader expects to be chained with the mdx loader.
+ */
+const loader = async function loader() {
+  const callback = this.async();
+
+  const exampleSource = this.resourcePath;
+  const { source } = findSource(exampleSource);
+
+  this.resolve('', source, (err, result) => {
+    const sourcePath = result || exampleSource;
+
+    if (result) {
+      // Add the src file to webpack's dependency list
```

Comment:
what types of file paths are we adding here and wouldn't already be in the dependency list?

Reply:
If you're including the lib file, but we're changing to include the src file, the src file wasn't required in the first place so webpack wouldn't know to add it to the dependencies it looks at for things like hot reloading.

Reply:
https://webpack.js.org/api/loaders/#thisadddependency